src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -373,10 +372,16 @@ internal static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, Nullable
                 return default;
             }
 
-            if (!nullableAnnotation.IsAnyNullable() && typeSymbol.IsNullableType())
+            switch (nullableAnnotation)
             {
-                // int?, T? where T : struct (add annotation)
-                nullableAnnotation = NullableAnnotation.Annotated;
+                case NullableAnnotation.Unknown:
+                case NullableAnnotation.NotAnnotated:
```

Comment:
>case NullableAnnotation.NotAnnotated: [](start = 16, length = 37)

Would it make sense to add an assert to TypeSymbolWithAnnotations's constructor that ```Unknown``` and ```NotAnnotated``` are never combined with a ```Nullable<T>```?